team-manager/src/test/unit/java/ua/knu/knudev/teammanager/service/DepartmentServiceTest.java

```diff
@@ -151,4 +157,159 @@ void should_throwDepartmentException_When_DepartmentDoesNotExistOnGet() {
 
         verify(departmentRepository, times(1)).findById(TEST_DEPARTMENT_ID);
     }
+
+    @Test
+    @DisplayName("Should create department when given valid data")
+    public void should_createDepartment_When_givenValidData() {
+        SpecialtyCreationDto specialtyCreationDto1 = new SpecialtyCreationDto(14.378, "Computer Engineering");
+        SpecialtyCreationDto specialtyCreationDto2 = new SpecialtyCreationDto(14.378, "Radio technics");
+        SpecialtyCreationDto specialtyCreationDto3 = new SpecialtyCreationDto(14.378, "History");
+
+        DepartmentCreationRequest departmentCreationRequest = buildDepartmentCreationRequest(
+                Set.of(specialtyCreationDto1, specialtyCreationDto2, specialtyCreationDto3),
+                TEST_DEPARTMENT_NAME
+        );
+
+        departmentService.createDepartment(departmentCreationRequest);
+
+        ArgumentCaptor<Department> departmentCaptor = ArgumentCaptor.forClass(Department.class);
+        verify(departmentRepository, times(1)).save(departmentCaptor.capture());
+
+        Department savedDepartment = departmentCaptor.getValue();
+
+        assertEquals(TEST_DEPARTMENT_NAME, savedDepartment.getName());
+        assertEquals(3, savedDepartment.getSpecialties().size());
+
+        assertTrue(savedDepartment.getSpecialties().stream()
+                .anyMatch(specialty -> "Computer Engineering".equals(specialty.getName())));
+        assertTrue(savedDepartment.getSpecialties().stream()
+                .anyMatch(specialty -> "Radio technics".equals(specialty.getName())));
+        assertTrue(savedDepartment.getSpecialties().stream()
```

Comment:
Instead of 3 separate asserts. Make 1 assert with allMatch. And for later reasosns, put this 3 specialties names to enums in class `DepartmentTestsConstants`. And it will be better to rename it `AcademicUnitsTestsConstants`

Reply:
done